app/services/assets/handlers/java.js

```diff
@@ -0,0 +1,163 @@
+import BaseCodeHandler from './baseCode';
+import Constants from '../../../constants/constants';
+
+const FILE_EXTENSION_LIST = ['java', 'class', 'jar'];
```

Comment:
I'm not sure that we want to scan the contents of class and JAR files, we would just want to try and match within the .java files.